### PR TITLE
Add default for `MixnetClientConfig::connection_pool_size`

### LINKED ISSUE #459: Add defaults to mixnect client config
Having to declare every option is annoying, let's define defaults for values where this can be done (e.g. `connection_pool_size`, `max_retries`, ..)

### PR DESCRIPTION
**Pull request by youngjoon-lee**
Closes #459 

@zeegomo 
As @al8n mentioned in https://github.com/logos-co/nomos-node/issues/459#issuecomment-1764760768, the default value of `max_retries` and `retry_delay` have been already defined, so that the default values can be used if nothing is set in the yaml. 
In addition to it, I defined the default of `connection_pool_size` as well.
I'm not sure if this is what you intended. Please let me know if you need something more.

### CODE REVIEW

**Review by al8n — APPROVED** [review]
LGTM!

File: mixnet/topology/src/lib.rs [hunk 1]
```diff
@@ -9,17 +9,17 @@ pub type MixnetNodeId = [u8; PUBLIC_KEY_SIZE];
 
 pub type Result<T> = core::result::Result<T, NymNodeRoutingAddressError>;
 
-#[derive(Serialize, Deserialize, Clone, Debug, Default)]
+#[derive(Serialize, Deserialize, Clone, Debug, Default, Eq, PartialEq)]
```
Comment (zeegomo) [inline mixnet/topology/src/lib.rs:12]: why this?

**Review by zeegomo — APPROVED** [review]
Mmm, maybe I was using an old ref, I'll check again later. Thanks!

File: mixnet/topology/src/lib.rs (hunk 1 above)
Comment (youngjoon-lee) [reply mixnet/topology/src/lib.rs:12]: I added this for the test that is added in this PR, in order to compare MixnetClientConfigs objects

Comment (youngjoon-lee) [reply mixnet/topology/src/lib.rs:12]: Ah, actually, we need only PartialEq (not Eq) for the test.

**Pull request merged**